src/types/receiver.ts

```diff
@@ -10,10 +10,9 @@ export interface ReceiverEvent {
   respond?: RespondFn;
 }
 
-// TODO: should it really be a requirement that on() returns this?
 export interface Receiver {
-  on(event: 'message', listener: (event: ReceiverEvent) => void): this;
-  on(event: 'error', listener: (error: Error | ReceiverAckTimeoutError) => void): this;
+  on(event: 'message', listener: (event: ReceiverEvent) => void): unknown;
```

Comment:
this is a notable change. this makes the requirements for someone implementing a `Receiver` looser - so that they don't necessarily have to return `this` from the `on()` method (but if they are inheriting from `EventEmitter` they would get that for free).